components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -265,10 +266,15 @@ class SitePermissionsFeature(
 
     private fun handleRuledFlow(
         permissionRequest: PermissionRequest,
```

Comment:
The function remain as it was, just adding an extra validation for when we have an autoplay request and we don't have `sitePermissionsRules` .
 
```kotlin
    private fun handleRuledFlow(
            permissionRequest: PermissionRequest,
            session: Session
    ): SitePermissionsDialogFragment? {
        //  For now we only support autoplay via sitePermissionsRules until we add support  for autoplay for specific sites see issue  ###
        return if (permissionRequest.isForAutoplay() && sitePermissionsRules == null) {
            permissionRequest.reject()
            session.contentPermissionRequest.consume { true }
            null
        } else {
            val action = requireNotNull(sitePermissionsRules).getActionFrom(permissionRequest)
            when (action) {
                SitePermissionsRules.Action.ALLOWED -> {
                    permissionRequest.grant()
                    session.contentPermissionRequest.consume { true }
                    null
                }
                SitePermissionsRules.Action.BLOCKED -> {
                    permissionRequest.reject()
                    session.contentPermissionRequest.consume { true }
                    null
                }
                SitePermissionsRules.Action.ASK_TO_ALLOW -> {
                    createPrompt(permissionRequest, session)
                }
            }
        }
    }
```

Reply:
Also it will be good to add a reference to ticket where we are going to add support for specific sites  autoplay request

Reply:
I feel like the control flow is becoming pretty confusing here, and it'll stay that way even after we make the eventual autoplay changes.  But we're short of time and we've probably already spent too much on this PR, so I'll take your advice and keep it close to what's in master.

Reply:
After we add support for adding autoplay settings per site, we can delete the part below:

```kotlin
        return if (permissionRequest.isForAutoplay() && sitePermissionsRules == null) {
            permissionRequest.reject()
            session.contentPermissionRequest.consume { true }
            null
        } else {
```
As autoplay is going to behave the same way as other permissions, with the only difference in `ASK_TO_ALLOW`

```kotlin
                SitePermissionsRules.Action.ASK_TO_ALLOW -> {
                   if(!permissionRequest.isForAutoplay()){
                       createPrompt(permissionRequest, session)
                    }else{
                      notifyAutoplaydelagate()  // TBD  
                      null
                  }   
                }
```